testnet/stacks-node/src/neon_node.rs

```diff
@@ -1586,11 +1618,28 @@ impl InitializedNeonNode {
             );
             for prev_block in last_mined_blocks.iter() {
                 debug!(
-                    "Consider in-flight Stacks tip {}/{} in {}",
+                    "Consider in-flight block {} on Stacks tip {}/{} in {} with {} txs",
+                    &prev_block.anchored_block.block_hash(),
                     &prev_block.parent_consensus_hash,
                     &prev_block.anchored_block.header.parent_block,
-                    &prev_block.my_burn_hash
+                    &prev_block.my_burn_hash,
+                    &prev_block.anchored_block.txs.len()
                 );
+                if prev_block.anchored_block.txs.len() == 1 {
+                    if last_mined_blocks.len() == 1 {
```

Comment:
These two checks are sufficient.  If there is only one transaction in this block, and there has only been one block mined, then it means that the miner only ran tenure once and it resulted in an empty block (i.e. because it didn't have time to actually consider any transactions).